src/libraries/Microsoft.Extensions.Configuration.Json/src/JsonConfigurationFileParser.cs

```diff
@@ -98,14 +97,15 @@ private void VisitValue(JsonElement value)
 
         private void EnterContext(string context)
         {
-            _context.Push(context);
-            _currentPath = ConfigurationPath.Combine(_context.Reverse());
+            _currentPath = _paths.Count > 0 ?
+                _paths.Peek() + ConfigurationPath.KeyDelimiter + context :
+                context;
+            _paths.Push(_currentPath);
         }
 
         private void ExitContext()
         {
-            _context.Pop();
-            _currentPath = ConfigurationPath.Combine(_context.Reverse());
+            _currentPath = _paths.Pop();
```

Comment:
If I am reading the code correctly, `_currentPath` is a cache for the top of the stack. This code should then be:
```
_paths.Pop();
_paths.TryPeek(out _currentPath);
```

Reply:
Ugh.  I'll improve the tests.

Reply:
Turns it out wasn't failing the tests because _currentPath is only ever used immediately after EnterContext is called, overwriting this invalid value.  I'll fix this up, though; _currentPath isn't actually needed at all.